packages/http-client-java/generator/http-client-generator-core/src/test/java/com/microsoft/typespec/http/client/generator/core/util/ClassNameUtilTests.java

```diff
@@ -8,32 +8,53 @@
+                + name + ".java").length());
 
         name = ClassNameUtil.truncateClassName("com.azure.resourcemanager.recoveryservicessiterecovery",
-            "src/tests/java", "com.azure.resourcemanager.recoveryservicessiterecovery.generated",
-            "InMageRcmUpdateApplianceForReplicationProtectedItemInputTests");
-        Assertions.assertEquals(
-            // workaround spelling check
-            "InMageRcmUpdateApplianceForReplicationProtectedItemInputTests".substring(0, 49), name);
+            "src/test/java", "com.azure.resourcemanager.recoveryservicessiterecovery.generated",
+            "InMageRcmUpdateApplianceForReplicationProtectedItemInput", "Tests");
+        Assertions.assertEquals(maxFileLength,
+            ("sdk/recoveryservicessiterecovery/azure-resourcemanager-recoveryservicessiterecovery/src/test/java/com/azure/resourcemanager/recoveryservicessiterecovery/generated/"
+                + name + ".java").length());
+
+        name = ClassNameUtil.truncateClassName(
+            "com.azure.resourcemanager." + KUBERNETES_CONFIGURATION + ".extensiontypes", "src/test/java",
+            "com.azure.resourcemanager." + KUBERNETES_CONFIGURATION + ".extensiontypes.generated",
+            "ExtensionTypesLocationGetWithResponse", "MockTests");
+        Assertions.assertEquals(maxFileLength,
+            ("sdk/" + KUBERNETES_CONFIGURATION + "/azure-resourcemanager-" + KUBERNETES_CONFIGURATION
+                + "-extensiontypes/src/test/java/com/azure/resourcemanager/" + KUBERNETES_CONFIGURATION
+                + "/extensiontypes/generated/" + name + ".java").length());
 
         // do nothing as too little remaining length for class name
         name = ClassNameUtil.truncateClassName("com.azure.resourcemanager.deviceprovisioningservicespadpadpadpadpadpad",
             "src/samples/java", "com.azure.resourcemanager.deviceprovisioningservicespadpadpadpadpadpad.generated",
-            "IotDpsResourceCheckProvisioningServiceNameAvailabilitySamples");
+            "IotDpsResourceCheckProvisioningServiceNameAvailability", "Samples");
         Assertions.assertEquals("IotDpsResourceCheckProvisioningServiceNameAvailabilitySamples", name);
 
         // no change
         name = ClassNameUtil.truncateClassName("com.azure.resourcemanager.datafactory", "src/samples/java",
-            "com.azure.resourcemanager.datafactory.generated", "DataFlowDebugSessionAddDataFlowSamples");
+            "com.azure.resourcemanager.datafactory.generated", "DataFlowDebugSessionAddDataFlow", "Samples");
         Assertions.assertEquals("DataFlowDebugSessionAddDataFlowSamples", name);
     }
+
+    @Test
+    public void testGetDirectoryNameForGraalVmConfig() {
+        String directoryName = ClassNameUtil.getDirectoryNameForGraalVmConfig("com.azure.resourcemanager",
+            "azure-resourcemanager-" + KUBERNETES_CONFIGURATION + "-extensiontypes");
```

Comment:
Would you help verify "fluxconfigurations"? A local run would suffice.

Reply:
generate and added to same test PR https://github.com/Azure/azure-sdk-for-java/pull/45388

your suppress is included as otherwise checkstyle would fail

Reply:
There is a CI issue in suppression PR, raised in https://github.com/Azure/azure-sdk-for-java/issues/45389